memory/ndctl.py

```diff
@@ -163,21 +205,37 @@ def setUp(self):
                         self.smm.install(pkg):
                     self.cancel('%s is needed for the test to be run' % pkg)
 
+            # Using fetch_asset() API
+            '''
             git_branch = self.params.get('git_branch', default='pending')
-            location = "https://github.com/pmem/ndctl/archive/"
-            location = location + git_branch + ".zip"
-            tarball = self.fetch_asset("ndctl.zip", locations=location,
-                                       expire='7d')
+            #location = location + git_branch + ".zip"
+            location = "https://github.com/pmem/ndctl/archive/refs/heads/pending.zip"
+            tarball = self.fetch_asset("ndctl.zip", locations=location, expire='7d')
             archive.extract(tarball, self.teststmpdir)
             os.chdir("%s/ndctl-%s" % (self.teststmpdir, git_branch))
-            process.run('./autogen.sh', sudo=True, shell=True)
-            process.run("./configure CFLAGS='-g -O2' --prefix=/usr "
-                        "--disable-docs "
-                        "--sysconfdir=/etc --libdir="
-                        "/usr/lib64", shell=True, sudo=True)
-            build.make(".")
-            self.ndctl = os.path.abspath('./ndctl/ndctl')
-            self.daxctl = os.path.abspath('./daxctl/daxctl')
+            '''
+
+            # Using get_repo() API
+            # Clone the branch
+            git.get_repo(url, branch='pending', destination_dir=self.teststmpdir)
+            os.chdir(self.teststmpdir)
+
+            if ndctl_project_version:
+                ndctl_tag_name = "v" + ndctl_project_version
+
+                # Checkout the desired tag
+                git_helper = GitRepoHelper(url, destination_dir=self.teststmpdir)
+                git_helper.checkout(branch=ndctl_tag_name, commit=None)
+
+                if (float(ndctl_project_version) < 73):
+                    self.autotools_build_system()                
+                else:
+                    self.meson_build_system(deps)   
+
+            # default to the meson way of building ndctl library
+            else:
+                self.meson_build_system(deps)
```

Comment:
move 
```
 git.get_repo(url, branch='pending', destination_dir=self.teststmpdir)
  os.chdir(self.teststmpdir)
 ```
 here as only in case of `ndctl_project_version` not given we can go with branch clonning

Reply:
ah some confusion as we need clone in case 2 as  `ndctl_project_version is not provided by the user` as in case 1 we already done tag checkout

Reply:
A call to get_repo() will always clone the repository with the latest version available. If this is being achieved then I guess, we do not have a necessity to have a call again to git_repo().

I mean, if we decide to go by what you are saying - the code will look something like below -
----------------------------------------------------------------------------------------------------------
if (upstream)
    - **git_repo()  --- will always Clone the latest available version of ndctl.**
    - change the directory.
    if (ndctl_project_version is given):
          - Clone the repo ( we have already done it above , so not required)
          - Checkout using the checkout() API
    else:
          **- Call git_repo() --> This would be a repetition!**
          - Call meson_build_system()  
----------------------------------------------------------------------------------------------------------

Reply:
yes, it serves the purpose but `git_repo() --- will always Clone the latest available version of ndctl.` clone specific branch that can be confusing so it is good we do operation where it relevant as a user confused with branch as well tag

Reply:
if (upstream)
if (ndctl_project_version is given):
- Clone the repo 
- Checkout using the checkout() API
else:
if (upstream)
if (ndctl_project_version is given):
- Clone the repo ( we have already done it above , so not required)
- Checkout using the checkout() API
else:
- git_repo() 
- Call meson_build_system()